src/ngraph/op/experimental/quantized_dot.cpp

```diff
@@ -39,13 +39,29 @@ op::QuantizedDot::QuantizedDot(const shared_ptr<Node>& data,
     auto& weights_shape = weights->get_shape();
     // QuantizedDot does [n, ic] * [oc, ic] = [n, oc]
     NODE_VALIDATION_CHECK(this,
-                          data_shape.size() == 2 && weights_shape.size() == 2 &&
-                              data_shape[1] == weights_shape[1],
+                          data_shape.size() == 2 && weights_shape.size() == 2,
                           "only valid tensors of rank 2 supported. data shape ",
                           data_shape,
                           " weights shape ",
                           weights_shape);
 
+    // Mkldnn requirement. So skip when it goes to reference::dot
```

Comment:
Can we handle this in CPU assignment?  Why are we encoding mkldnn requirements at the op level?

Reply:
The checks/output shape set were already mkldnn specific. I added check for the output shape to be a conventional [m, k] for a [m,n]*[n,k].  This op seems to assume we only have a mkldnn implementation, which was true before this PR

Reply:
this op should not assume we only have an mkldnn implementation

Reply:
I still think it would be best to not have mkldnn specifics in the ops but I will not hold up the PR over it.

Reply:
We really need to get rid of backend specific code in core ngraph. It belongs in the cpu backend, not here.